packages/worker/src/workflow/threaded-vm.ts

```diff
@@ -66,6 +68,23 @@ export class WorkerThreadClient {
 
       completion.resolve(result.output);
     });
+    workerThread.on('exit', () => {
+      this.workerExited = true;
+      if (this.shutDownRequested) {
+        return; // ignore
+      }
+      const completions = this.requestIdToCompletion.values();
+      this.requestIdToCompletion = new Map();
+      for (const completion of completions) {
+        completion.reject(
+          new UnexpectedError(
```

Comment:
Sorry I am unfamiliar with how the threads work here. Is there no way we can capture an error thrown from an async workflow instead of having it terminate a worker thread? Can the workflow function not be wrapped in a try+await+catch _inside_ the thread? If concerned about top-level code throwing, can the entire `require` be wrapped in a try+await+catch?

Reply:
We terminate the thread if it we can't determine the context that threw the handled error to avoid accidentally completing the activation successfully.
There's no way to catch these errors unfortunately.